File: songs/go/myutils/myutils.go
package myutils

func exclude(value string, tags []string) []string {
	var newTags []string

	for _, tag := range tags {
		if tag != value {
			newTags = append(newTags, tag)
		}
	}

	return newTags
}

func Unique(tags []string) []string {
	if len(tags) <= 1 {
		return tags
	} else {
		head := tags[0]
		tail := tags[1:len(tags)]

		newHeads := []string{head}
		newTails := Unique(exclude(head, tail))

		return append(newHeads, newTails...)
	}
}

File: songs/rust/my_utils.rs
pub fn unique<T: Clone + PartialEq>(
    tags: Vec<T>
) -> Vec<T> {
    if tags.len() <= 1 {
        tags
    } else {
        let head: T = tags[0].clone();
        let tail: Vec<T> = tags[1..].to_vec();

        let mut excluded: Vec<T> = Vec::new();

        for tag in tail.into_iter() {
            if tag != head {
                excluded.push(tag); 
            }
        }    

        let mut new_tail = unique(excluded);

        new_tail.insert(0, head);
        new_tail
    }
}
